RFCs/0011-Test-Session-Timeout.md

```diff
@@ -0,0 +1,51 @@
+# 0011 Test Session Timeout
+
+## Summary
+Allow user to specify timeout which will terminate a test session when it exceeds a given timeout.
+
+## Motivation
+The test platform should have the ability for users to specify a test run level timeout which allows users to terminate a test session when it exceeds a given timeout. This ensures that resources are well consumed and test sessions are constrained to a set time.
+
+## Detailed Design
+
+User would specify timeout using runsetting or through commandline.
+
+### User can specify timeout using runsettings as follows:
+```xml
+<Runsettings>
+  <RunConfiguration>
+     <!-- Specify timeout in milliseconds. A valid value should be greater than 0 -->
+     <TestSessionTimeout>10000</TestSessionTimeout>
+  </RunConfiguration>
+</Runsettings>
+```
+
+### User can specify timeout using commandline as follows:
+```
+vstest.console <testContainersList> -- RunConfiguration.TestSessionTimeout=10000
+```
+
+### Overview of changes
+Testplatform will cancel the current test run if it has exceeded given `TestSessionTimeout`. It will also make sure that testhost is getting test cancel request. Finally it will conclude the run with whatever test run till that point.
+
+### Exit Code of vstest.console
+It will be a failed test and exit code will be 1
+
+### Console output
```

Comment:
Tagging @pvlakshm for review.

@Faizan2304 please update configure.md in this PR? This will be supported for 15.5 onwards?